osidb/models.py

```diff
@@ -112,7 +112,71 @@ class FlawType(models.TextChoices):
     WEAKNESS = "WEAKNESS"
 
 
-class FlawImpact(models.TextChoices):
+class ComparableTextChoices(models.TextChoices):
+    """
+    extension of the models.TextChoices classes
+    making them comparable with the standard operators
+
+    the comparison order is defined simply by the
+    top-down order in which the choices are written
+    """
+
+    @classmethod
+    def get_choices(cls):
+        """
+        get processed choices
+        """
+        return [choice[0] for choice in cls.choices]
+
+    @property
+    def weight(self):
+        """
+        weight of the instance for the comparison
+        defined by the order of the definition of the choices
+        """
+        return self.get_choices().index(str(self))
+
+    def _check_type(self, other):
+        """
+        ensure that that we are comparing the instances of the same type as
+        comparing different types (even two ComparableTextChoices) is undefined
+        """
+        if type(self) != type(other):
+            raise NotImplementedError(
+                f"Comparison of {type(self)} and {type(other)} is undefined"
+            )
+
+    def __hash__(self):
+        return super().__hash__()
+
+    def __eq__(self, other):
+        # let us only prevent equal comparison to different ComparableTextChoices
+        # as Django internally uses it heavily (probably together with some conversions)
+        if isinstance(other, ComparableTextChoices):
+            self._check_type(other)
+        return super().__eq__(other)
```

Comment:
we don't need to override `__eq__()` as the default is identity comparison (e.g. `self is other`), this means that you also don't have to override `__hash__()`

Reply:
Without overriding this the result is simply `False` which might leave the programmer in a false conviction that (s)he is actually not making a programming error comparing apples and oranges but the objects are simply comparable but not the same. I think this contradicts what you suggested before which was to prevent comparing apples and oranges. So I would like to understand whether your are OK with comparing apples and oranges or not.

Reply:
Yes, maybe I contradicted myself and took the usage of the word error liberally, but I showed a clear example of how it's supposed to be done.

If we compare Enum A to Enum B the result of that comparison should always be `False` or `NotImplemented` because they're not comparable even if they might have similar values defined.

We could raise an error but since we don't control the base object (`TextChoices`) nor the systems that interface with it (Django, as you point out in your comment) then we should stick to the base API, which is to return `NotImplemented` if the objects are not comparable. This is done so that we don't inadvertently break parts of Django, and it's also the reason why I was originally against this approach, because we just don't know how it might affect the rest of the application, especially if you're not careful about the implementation.

Reply:
I am OK with replace `raise ...` with `return NotImplemented` however then in all cases including `__eq__` so it is consistent. I will adjust the implementation.